File: Visualization/WebUI/src/app/interfaces/console-log.ts
import { LogType, ConsoleColor, LogTypeToColorMap } from './utilities';

export class ConsoleLog {
    message: string;
    
    timeStamp: string;
    
    logType: LogType;

    public constructor(message: string, timeStamp: string, logType: LogType) {
        this.message = message;
        this.timeStamp = timeStamp;
        this.logType = logType;
    }

    getConsoleColor(): ConsoleColor {
        return LogTypeToColorMap.get(this.logType);
    };
}

File: Common/Common.Models/ConsoleLog.cs
namespace Common.Models
{
    using System;
    using Common.Models.Enums;

    public class ConsoleLog
    {
        public string Message { get; set; }

        public DateTime TimeStamp { get; } = DateTime.Now;

        public LogType Type { get; set; }

        public ConsoleLog() { }

        public ConsoleLog(string message, LogType type)
        {
            this.Message = message;
            this.Type = type;
        }
    }
}
